cmd/bootstrapper/server.go

```diff
@@ -134,8 +134,40 @@ func (s *Server) Start(ctx context.Context, errCh chan error, params *NetworkPar
 		// start generating fallback data
 		s.eg.Go(
 			func() error {
-				s.genDataLoop(ctx, errCh, last, params.updateActiveSetTime, s.GenFallbackActiveSet)
-				return nil
+				var (
+					errs    = 0
+					maxErrs = 10
+					timer   *time.Timer
+					timeC   <-chan time.Time
```

Comment:
Why is this channel needed? Would it be possible to just select on `case <-timer.C`?

Reply:
Because the channel is sometimes `nil`, since it triggers a retry only whenever there's an error. If there's no errors, you don't want to trigger a retry, therefore the channel is `nil`. Due to this, you can't select on the timer channel directly, because at the beginning, the timer is not initialized, causing a panic.

Reply:
? Reading a `nil` channel doesn't panic: https://go.dev/play/p/AVS1DOGWtrb

Reply:
`timer` is not initialized in the first round of retries. When you create a new timer, it already starts the goroutine that will expire it. So you can't select on `timer.C` when `timer` (which is a pointer to a timer) is `nil`